hazelcast/src/main/java/com/hazelcast/cardinality/impl/CardinalityEstimatorService.java

```diff
@@ -117,27 +117,26 @@ public Operation prepareReplicationOperation(PartitionReplicationEvent event) {
     public void commitMigration(PartitionMigrationEvent event) {
         if (event.getMigrationEndpoint() == MigrationEndpoint.SOURCE) {
             int thresholdReplicaIndex = event.getNewReplicaIndex();
-            if (thresholdReplicaIndex == -1 || thresholdReplicaIndex > 1) {
-                clearPartitionReplica(event.getPartitionId());
-            }
+            clearPartitionReplica(event.getPartitionId(), thresholdReplicaIndex);
         }
     }
 
     @Override
     public void rollbackMigration(PartitionMigrationEvent event) {
         if (event.getMigrationEndpoint() == MigrationEndpoint.DESTINATION) {
             int thresholdReplicaIndex = event.getCurrentReplicaIndex();
```

Comment:
I see that it also named this way at other services. Ok.. forget my previous comment. Perhaps the name of the field is just wrong :)